arch/arm/core/thread.c

```diff
@@ -13,8 +13,7 @@
  */
 
 #include <kernel.h>
-#include <toolchain.h>
-#include <kernel_structs.h>
+#include <ksched.h>
```

Comment:
Why is this needed, actually? Couldn't find it in the source code :)

Reply:
`Z_ASSERT_VALID_PRIO` is defined in `ksched.h`.